### PR TITLE
Fetch blog posts from sanity

### PR DESCRIPTION
**Pull request by jfranciscosousa**
We want to migrate and have all of our content on Sanity so all of our little side projects can reference that entirely, without dealing with local files.

This PR creates a data file on `_data` that lists every single blog post.

Also removed the syntax highlighting plugin and instead apply a plugin to `markdown-it`, also change the filters a little bit.

I also changed the image optimizer so it deals with remote images. It also caches the remote image so we only download it once.

### CODE REVIEW

**Review by zediogoviana — APPROVED** [review]
Looks good to me 👍 
But I'd wait on a review by someone else, as this is a somewhat big change

File: site/_data/blog_posts.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const { request, gql } = require("graphql-request");
+
+const API_URL =
+  "https://5aprln8a.apicdn.sanity.io/v1/graphql/production/default";
```
Comment (DavideSilva) [inline site/_data/blog_posts.js]: I'm guessing there isn't but just want to check if there is a problem to having this URL be publicly available?

**Review by davelange — APPROVED** [review]
Looks good :+1:

File: site/_data/blog_posts.js (hunk 1 above)
Comment (jfranciscosousa) [reply site/_data/blog_posts.js]: We can hide it just to make it more difficult for people to target denial of wallet attacks but this URL can be easily inferred.

Comment (jfranciscosousa) [reply site/_data/blog_posts.js]: Yes, you can get this URL from the images CDN. It's supposed to be publicly available.

Comment (DavideSilva) [reply site/_data/blog_posts.js]: Still, this should probably be set in a environment variable. If not for security purposes, at least for convenience if we need to change it on a later date

Comment (jfranciscosousa) [reply site/_data/blog_posts.js]: This will make it harder for people to check out the code and run it locally, but I'll add it anyway.

**Review by DavideSilva — APPROVED** [review]

**Pull request merged**